app/src/main/java/com/babylon/wallet/android/domain/usecases/ResolveAccountsLedgerStateUseCase.kt

```diff
@@ -0,0 +1,86 @@
+package com.babylon.wallet.android.domain.usecases
+
+import com.babylon.wallet.android.data.gateway.apis.StateApi
+import com.babylon.wallet.android.data.gateway.extensions.defaultDepositRule
+import com.babylon.wallet.android.data.gateway.extensions.isEntityActive
+import com.babylon.wallet.android.data.gateway.extensions.toProfileDepositRule
+import com.babylon.wallet.android.data.gateway.generated.models.DefaultDepositRule
+import com.babylon.wallet.android.data.gateway.generated.models.StateEntityDetailsOptIns
+import com.babylon.wallet.android.data.gateway.generated.models.StateEntityDetailsRequest
+import com.babylon.wallet.android.data.gateway.model.ExplicitMetadataKey
+import com.babylon.wallet.android.data.repository.toResult
+import com.babylon.wallet.android.di.JsonConverterFactory
+import com.babylon.wallet.android.di.ShortTimeoutGatewayHttpClient
+import com.babylon.wallet.android.di.buildApi
+import com.babylon.wallet.android.domain.model.AccountWithOnLedgerStatus
+import okhttp3.OkHttpClient
+import rdx.works.profile.data.model.apppreferences.Radix
+import rdx.works.profile.data.model.currentGateway
+import rdx.works.profile.data.model.pernetwork.Network
+import rdx.works.profile.data.repository.ProfileRepository
+import retrofit2.Converter
+import javax.inject.Inject
+
+class ResolveAccountsLedgerStateUseCase @Inject constructor(
+    private val profileRepository: ProfileRepository,
+    @ShortTimeoutGatewayHttpClient private val httpClient: OkHttpClient,
```

Comment:
Why we need shorter timeout on this HttpClient ?

Reply:
That was the idea Cyon brought somewhere, so that we don't end up creating account for very long time, if we fail to quickly sync the state, we use default values